Bio/Seq.py

```diff
@@ -2385,6 +2385,29 @@ def _translate_str(sequence, table, stop_symbol="*", to_stop=False,
         valid_letters = set(IUPAC.ambiguous_dna.letters.upper() +
                             IUPAC.ambiguous_rna.letters.upper())
     n = len(sequence)
+
+    # Check for tables with 'ambiguous' (dual-coding) stop codons:
+    for codon in stop_codons:
+            try:
+                forward_table[codon]
+                if to_stop:
+                    raise ValueError("This table contains at least one stop "
+                                     "codon ('{}') which codes for both "
+                                     "'STOP' and an amino acid ('{}'). You "
+                                     "can not use 'to_stop=True' with this "
+                                     "table."
+                                     .format(codon, forward_table[codon]))
+                else:
+                    warnings.warn("This table contains at least one stop "
+                                  "codon ('{}') which codes for both 'STOP' "
+                                  "and an amino acid ('{}'). Such codons will "
+                                  "be translated as amino acid."
+                                  .format(codon, forward_table[codon]),
+                                  BiopythonWarning)
+                break
+            except KeyError:
+                pass  # KeyError is the expected behaviour for most tables
```

Comment:
This may be more of a style thing, but I often feel that `except <Exception>: pass` can be better rewritten to convey intent.

How about something like this?

```
for codon in stop_codons:
    if codon not in forward_table:
        continue
    if to_stop:
        <raise exception>
    <warn>
```

We can even remove the `continue` clause:

```
for codon in {c for c in stop_codons if c in forward_table}:
    if to_stop:
        <raise exception>
    <warn>
```

This would raise the exception on the first ambiguous codons if `to_stop` is `True`, but would raise warnings for all the ambiguous codons otherwise.

Reply:
Unfortunately, this doesn't work. ```codon in forward_table``` gives a ```TypeError``` for ambiguous tables, while ```forward_table[codon]``` works nicely (either returning an amino acid or raising a ```KeyError```, as expected). 
I don't know if this is a bug or intended, the ```__getitem__``` method in ```AmbiguousForwardTable``` is quite complex...
So if you don't insist, I would let it as it is.
Example:
```
>>> from Bio.Data import CodonTable
>>> table = CodonTable.ambiguous_dna_by_id[1]
>>> table.forward_table['ATG']
'M'
>>> 'ATG' in table.forward_table

Traceback (most recent call last):
  File "<pyshell#61>", line 1, in <module>
    'ATG' in table.forward_table
  File "C:\Program Files\Anaconda32\lib\site-packages\Bio\Data\CodonTable.py", line 375, in __getitem__
    self.ambiguous_nucleotide)
  File "C:\Program Files\Anaconda32\lib\site-packages\Bio\Data\CodonTable.py", line 209, in list_possible_proteins
    c1, c2, c3 = codon
TypeError: 'int' object is not iterable
```

Reply:
Aha, I had thought earlier `forward_table` should behave like a plain `dict`. That is apparently not the case.

The error we're seeing is because `forward_table` (or rather, its class) doesn't define a `__contains__` method, [which makes the `in` operation resort to using `__getitem__` but using integers instead of `ATG`](https://docs.python.org/3/reference/expressions.html#membership-test-details).

Without an expected `__contains__` behavior, I can understand that the `exception <E>: pass` style is the one that does what we want ~ so let's keep it that way for now.

Reply:
Given it is dictionary like, do you think is the expected behaviour of the missing ``__contains__``method clearly defined? If so we could add it?

Reply:
I'm thinking about adding it. I'll see how it works